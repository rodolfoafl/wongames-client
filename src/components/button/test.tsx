import { render, screen } from '@testing-library/react'

import Button from '.'
import { ShoppingCart } from 'lucide-react'

describe('<Button />', () => {
  it('should render the medium size button by default', () => {
    const { container } = render(<Button>Buy Now</Button>)

    expect(screen.getByRole('button', { name: /Buy Now/i })).toHaveClass(
      'h-10 px-4 text-sm'
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the small size button', () => {
    render(<Button size="small">Buy Now</Button>)

    expect(screen.getByRole('button', { name: /Buy Now/i })).toHaveClass(
      'h-[1.875rem] px-[0.875rem] text-xs'
    )
  })

  it('should render the large size button', () => {
    render(<Button size="large">Buy Now</Button>)

    expect(screen.getByRole('button', { name: /Buy Now/i })).toHaveClass(
      'h-[3.125rem] px-5 text-base'
    )
  })

  it('should render the full width button', () => {
    render(<Button fullWidth>Buy Now</Button>)

    expect(screen.getByRole('button', { name: /Buy Now/i })).toHaveClass(
      'w-full'
    )
  })

  it('should render the button with an icon ', () => {
    render(<Button icon={<ShoppingCart data-testid="icon" />}>Buy Now</Button>)

    expect(screen.getByText(/buy now/i)).toBeInTheDocument()
    expect(screen.getByTestId('icon')).toBeInTheDocument()
  })
})
