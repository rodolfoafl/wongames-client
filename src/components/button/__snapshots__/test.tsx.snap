// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the medium size button by default 1`] = `
<button
  class="flex items-center justify-center rounded-sm border-none bg-[linear-gradient(180deg,#FF5F5F_0%,#F062C0_100%)] cursor-pointer p-2 font-semibold text-white hover:bg-[linear-gradient(180deg,#E35565_0%,#D958A6_100%)] h-10 px-4 text-sm"
>
  <span>
    Buy Now
  </span>
</button>
`;
